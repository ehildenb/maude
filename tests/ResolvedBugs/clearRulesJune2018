#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < $srcdir/clearRulesJune2018.maude -no-banner -no-advise \
  > clearRulesJune2018.out 2>&1

diff $srcdir/clearRulesJune2018.expected clearRulesJune2018.out > /dev/null 2>&1
