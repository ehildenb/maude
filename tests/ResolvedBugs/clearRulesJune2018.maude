set show timing off .
set show advisories off .

***
***	Check that object-message rewriting is correctly initialized after
***	set clear rules off .
***

mod TEST is
  including CONFIGURATION .
  op m   : Oid Oid -> Msg [msg] .
  ops 1 2 : -> Oid .
  op User : -> Cid .

  var A : AttributeSet .
  vars X Y : Oid .

  rl < X : User | A > m(X, Y) => < X : User | A > m(Y, X) .
endm

set clear rules off .

frew [1] < 1 : User | none > m(1, 2) .
