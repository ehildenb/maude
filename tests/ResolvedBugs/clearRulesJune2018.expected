==========================================
frewrite [1] in TEST : m(1, 2) < 1 : User | none > .
rewrites: 1
result Configuration: m(2, 1) < 1 : User | none >
Bye.
